neurons/validators/utils/utils.py

```diff
@@ -32,4 +36,30 @@ def count_hotkeys_per_ip(filtered_axons):
         ip = axon.ip
         hotkey_count_per_ip[ip] = hotkey_count_per_ip.get(ip, 0) + 1
 
-    return hotkey_count_per_ip
\ No newline at end of file
+    return hotkey_count_per_ip
+
+def generate_challenge_to_check(node, start_block_height, last_block_height, k=20):
+    blocks_to_check = random.sample(range(start_block_height, last_block_height + 1), k=k)
```

Comment:
need to add a check here that last_block_height + 1 - start_block_height is > k otherwise it will raise an exception (see main branch)